src/apt-map.ts

```diff
@@ -1,218 +1,297 @@
-      mapped.set(mapKey, resultValue);
-    }
+    this.forEach((mapValue, mapKey) => {
+      const resultValue = cb(mapValue, mapKey, this)
 
-    return mapped;
+      mapped.set(mapKey, resultValue)
+    })
+
+    return mapped
   }
 
-  remapToArray<T = any>(cb: (value: V, key?: K, aptMap?: AptMap<K, V>) => T): T[] {
-    const mapped: T[] = [];
+  /**
+   * Re-maps an apt map to an array with values returned in callback function.
+   */
+  remapToArray<T = any>(cb: Callback<K, V, T>) {
+    const mapped: T[] = []
 
-    for (const [mapKey, mapValue] of this) {
-      const resultValue = cb.apply(this, [mapValue, mapKey, this]);
-      mapped.push(resultValue);
-    }
+    this.forEach((mapValue, mapKey) => {
+      const resultValue = cb(mapValue, mapKey, this)
+
+      mapped.push(resultValue)
+    })
 
-    return mapped;
+    return mapped
   }
 
-  remapToObject<T = any>(cb: (value: V, key?: K, aptMap?: AptMap<K, V>) => T): Record<string, T> {
-    const mapped: Record<string, T> = {};
+  /**
+   * Re-maps an apt map to an object keys from map and values returned in callback function.
+   */
+  remapToObject<T = any>(cb: Callback<K, V, T>) {
+    const mapped: Record<string, T> = {}
 
-    for (const [mapKey, mapValue] of this) {
+    this.forEach((mapValue, mapKey) => {
       const objectKey = typeof mapKey === 'object'
         ? JSON.stringify(mapKey)
         : String(mapKey)
 
-      mapped[objectKey] = cb.apply(this, [mapValue, mapKey, this]);
-    }
+      mapped[objectKey] = cb(mapValue, mapKey, this)
+    })
 
-    return mapped;
+    return mapped
   }
 
-  setMany<T extends V>(aptMap: AptMap<K, T>): AptMap<K, V> {
-    aptMap.forEach((value, key) => this.set(key, value));
-    return this;
+  /**
+   * Adds/updates entries specified as input apt map. Returns updated apt map.
+   */
+  setMany<T extends V>(aptMap: AptMap<K, T>) {
+    aptMap.forEach((value, key) => this.set(key, value))
+
+    return this
   }
 
-  some(cb: (value: V, key?: K, aptMap?: AptMap<K, V>) => boolean): boolean {
-    let result = false;
+  /**
+   * Returns `true` if a condition in callback function is satisfied at least for one entry, otherwise `false`
+   */
+  some(cb: Callback<K, V>) {
+    let result = false
 
-    for (const [mapKey, mapValue] of this) {
-      const isTruthy = cb.apply(this, [mapValue, mapKey, this]);
+    this.forEach((mapValue, mapKey) => {
+      const isTruthy = cb(mapValue, mapKey, this)
 
-      if (isTruthy) {
-        result = true;
-        break;
-      }
-    }
+      if (!isTruthy)
+        return
+
+      result = true
+    })
 
-    return result;
+    return result
   }
 
-  toObject(): Record<string, V> {
-    return this.remapToObject((value) => value);
+  /**
+   * Converts apt map to object.
+   */
+  toObject() {
+    return this.remapToObject(value => value)
   }
 
-  valuesAsArray(): V[] {
-    return Array.from(this.values());
+  /**
+   * Returns an array of values.
+   */
+  valuesAsArray() {
```

Comment:
why all them methods and getters lost return type

Reply:
This is type inferencing, hover over the function and see the unknown magic behind the scenes.

![image](https://github.com/jarcko/apt-map/assets/15153245/bfa0139b-2026-4353-98cd-86dbcea03043)